cwms_radar_api/src/main/java/cwms/radar/data/dao/RecentValueDao.java

```diff
@@ -0,0 +1,257 @@
+package cwms.radar.data.dao;
+
+import java.sql.Timestamp;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import cwms.radar.data.dto.Tsv;
+import cwms.radar.data.dto.TsvDqu;
+import cwms.radar.data.dto.TsvDquId;
+import cwms.radar.data.dto.TsvId;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.Condition;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.Record;
+import org.jooq.Record1;
+import org.jooq.Result;
+import org.jooq.ResultQuery;
+import org.jooq.SelectConditionStep;
+import org.jooq.SelectHavingStep;
+
+import usace.cwms.db.jooq.codegen.tables.AV_CWMS_TS_ID2;
+import usace.cwms.db.jooq.codegen.tables.AV_TSV;
+import usace.cwms.db.jooq.codegen.tables.AV_TSV_DQU;
+
+import static org.jooq.impl.DSL.asterisk;
+import static org.jooq.impl.DSL.field;
+import static org.jooq.impl.DSL.max;
+import static org.jooq.impl.DSL.partitionBy;
+
+public class RecentValueDao extends JooqDao<TsvDqu>
```

Comment:
This is clearly just timeseries operations, albeit slightly specific ones, does it really make sense to have a separate DAO or should this all just be lumped into the/a TimeSeriesDAO? (I'm not actually sure, just trying to start a discussion.)

Reply:
I just pulled the TimeSeries methods out of CwmsDataManager and combined them with these methods into a TimeSeriesDao object.   These queries were ported over from my cwms mobile queries and I'm not 100% sure I want to expose TsvDqu and Tsv objects but at the moment the recent methods aren't access by the controllers

Reply:
I would say that at the least, that shouldn't be the name. But couldn't disagree with the concept enough to say "no don't do that."